src/main/java/com/capitalone/dashboard/model/LibraryPolicyResult.java

```diff
@@ -114,6 +130,43 @@ public void addThreat(LibraryPolicyType type, LibraryPolicyThreatLevel level, St
         }
     }
 
+    public void addThreat(LibraryPolicyType type, LibraryPolicyThreatLevel level, LibraryPolicyThreatDisposition disposition, String component) {
+        Set<Threat> threatSet = threats.get(type);
+
+        if (CollectionUtils.isEmpty(threatSet)) {
+            Threat threat = new Threat(level, 1);
+            threat.getComponents().add(getComponentPlusDisposition(component,disposition));
+            threat.addDispositionCount(disposition);
+            Set<Threat> set = new HashSet<>();
+            set.add(threat);
+            threats.put(type, set);
```

Comment:
Can we use Collections.singleton(threat) i.e. threats.put(type, Collcetions.singleton(threat));